HRIS.Services/Services/EmployeeService.cs

```diff
@@ -122,45 +128,54 @@ public async Task<List<EmployeeDto>> GetAll(string userEmail = "")
 
     public async Task<EmployeeDto?> GetEmployee(string email)
     {
+        var emailExists = await CheckUserEmailExist(email);
+        if (!emailExists)
+            throw new CustomException("User email not found");
+
         var employee = await _db.Employee
                                 .Get(employee => employee.Email == email)
                                 .AsNoTracking()
                                 .Include(employee => employee.EmployeeType)
                                 .Include(employee => employee.PhysicalAddress)
                                 .Include(employee => employee.PostalAddress)
                                 .Select(employee => employee.ToDto())
-                                .Take(1)
-                                .FirstOrDefaultAsync();
+                                .FirstOrDefaultAsync() ?? throw new CustomException("Unable to Load Employee");
 
-        if (employee == null)
-            throw new CustomException("Unable to Load Employee");
+        if (_identity.IsSupport == false && _identity.EmployeeId != employee.Id)
+            throw new CustomException("Unauthorized Access");
 
         return employee;
     }
 
     public async Task<EmployeeDto> GetEmployeeById(int id)
     {
-        return await _db.Employee
+        if (_identity.IsSupport == false && _identity.EmployeeId != id)
+            throw new CustomException("Unauthorized Access");
+
+        var employee = await _db.Employee
                                 .Get(employee => employee.Id == id)
                                 .AsNoTracking()
                                 .Include(employee => employee.EmployeeType)
                                 .Include(employee => employee.PhysicalAddress)
                                 .Include(employee => employee.PostalAddress)
                                 .Select(employee => employee.ToDto())
                                 .FirstOrDefaultAsync() ?? throw new CustomException("Unable to Load Employee");
+
+        return employee;
     }
 
     public async Task<EmployeeDto> UpdateEmployee(EmployeeDto employeeDto, string email)
     {
```

Comment:
Logic: is string email actually being used in the service, controller and front end or are we using employeeDto.email?